### PR TITLE
feat: new config format and parser

### LINKED ISSUE #392: Better config management
#### Reason/Context
When I want to add something to the config file currently I have to take the following steps:
1) update the `lib/index.d.ts` and add my config types.
2) update the `docs/config-file.md` to reflect the new configs that I have added/changed.
3) If no config is provided in my config file I have to hardcode the default values in adapters themselves.

When working with the codebase, it's a little hard to go back and forth between these files and make sure the are synced and the parsing logic is correct. 

#### Description
IMO we can make some improvements to the codebase and make it easier for the new contributors to understand the codebase by having a config class that is responsible for all three of those above.

1) The class should be documented correctly with `jsdoc` so we can generate the `doc/config-file.md` from it instead of maintaining it separately.

2) all of the parsing logic should be handled by this class itself. I shouldn't have to look around for the port of my server in three different places and finally hardcode the default value in each `Adapter`. 😄 

3) Providing default values for each configuration.

### PR DESCRIPTION
**Pull request by Souvikns**
**Description**

Since we are moving the client auth from the `config file` to `auth` folder we need to update our config file format and update the parser as well to fetch these data differently. 

This PR is trying to achieve two things:

- [ ] New format for the config file (also the requirements for the config file now that auth parameters are removed)
- [ ] Update the config file parsing logic according the the new format 

**Related issue(s)**

Fixes #392

### CODE REVIEW

File: docs/config-file.md [hunk 1]
```diff
@@ -18,11 +18,9 @@ This function must return an object with the following shape:
...
-    kafka: {},
-    websocket: {},
-    mqtt: {},
-    cluster: {},
-    http: {}
+    docs: {},
+    server: {},
+    cluster: {}
   }
 }
```
Comment (Souvikns) [inline docs/config-file.md:25]: [This](https://github.com/asyncapi/glee/blob/b0c1670f777ad2a4dd0e33004d5ab78a72964463/docs/config-file.md) was the old format for `glee-config` and it was done to configure WebSocket servers for socket-io or pass in a custom server object. So building on that I am trying to add a new object named `server` where we would pass in the custom server config. The parser would take a server name from the spec and see if any config exists in the config file and then load that accordingly.

**Review by Souvikns — COMMENTED** [review]
@KhudaDad414 I am trying to change the format for the `config-file` now that the `auth` parameters are being removed.

File: docs/config-file.md (hunk 1 above)
Comment (KhudaDad414) [reply docs/config-file.md:25]: It looks good, the only thing that worries me a little about this approach is writing an interface for it. I mean we want type suggestion/safety on the file right? with server name as a key I am not sure how we can write protocol specific interfaces. 🤔

Comment (Souvikns) [reply docs/config-file.md:25]: I think we can still achieve that, I am not sure but I used notion API which had something like this kind of type 

```ts
interface T {
  server: {
    [name:string]: {
      httpAdapter: any,
      adapter: string,
      port: number
    }
  }
}

```

Comment (KhudaDad414) [reply docs/config-file.md:25]: yeah, If type can be checked then let's do this. 👍

**Comment by KhudaDad414** [comment]
@Souvikns what are we going to do with this? 🤔

**Comment by Souvikns** [comment]
Since the config file has changed a lot, I am closing this and will open a new PR.

**Pull request closed without merging**